tests/common/test_cgroups.py

```diff
@@ -304,6 +321,129 @@ def test_limits(self):
+
+        self.cleanup_cgroup_telemetry()
+
+    @patch("azurelinuxagent.common.osutil.default.DefaultOSUtil.get_total_cpu_ticks_since_boot")
+    @patch("azurelinuxagent.common.cgroup.CpuCgroup._get_current_cpu_total")
+    @patch("azurelinuxagent.common.cgroup.CpuCgroup._update_cpu_data")
+    def test_telemetry_calculations(self, *args):
+        self.cleanup_cgroup_telemetry()
+
+        num_polls = 10
+        num_extensions = 1
+        num_summarization_values = 7
+
+        cpu_percent_values = [random.randint(0, 100) for _ in range(num_polls)]
+
+        # only verifying calculations and not validity of the values.
+        memory_usage_values = [random.randint(0, 8 * 1024 ** 3) for _ in range(num_polls)]
+        max_memory_usage_values = [random.randint(0, 8 * 1024 ** 3) for _ in range(num_polls)]
+
+        for i in range(num_extensions):
+            dummy_cpu_cgroup = CpuCgroup("dummy_extension_{0}".format(i), "dummy_path")
+            CGroupsTelemetry.track_cgroup(dummy_cpu_cgroup)
+
+            dummy_memory_cgroup = MemoryCgroup("dummy_extension_{0}".format(i), "dummy_path")
+            CGroupsTelemetry.track_cgroup(dummy_memory_cgroup)
+
+        self.assertEqual(2 * num_extensions, len(CGroupsTelemetry._tracked))
+
+        with patch("azurelinuxagent.common.cgroup.CpuCgroup._get_cpu_percent") as patch_get_cpu_percent:
+            with patch(
+                    "azurelinuxagent.common.cgroup.MemoryCgroup._get_memory_max_usage") as patch_get_memory_max_usage:
+                with patch("azurelinuxagent.common.cgroup.MemoryCgroup._get_memory_usage") as patch_get_memory_usage:
+                    with patch("azurelinuxagent.common.cgroup.CGroup.is_active") as patch_is_active:
+                        for i in range(num_polls):
+                            patch_is_active.return_value = True
+                            patch_get_cpu_percent.return_value = cpu_percent_values[i]
+                            patch_get_memory_usage.return_value = memory_usage_values[i]  # example 200 MB
+                            patch_get_memory_max_usage.return_value = max_memory_usage_values[i]  # example 450 MB
+                            CGroupsTelemetry.poll_all_tracked()
+
+        collected_metrics = CGroupsTelemetry.collect_all_tracked()
+        for i in range(num_extensions):
+            name = "dummy_extension_{0}".format(i)
+
+            self.assertIn(name, collected_metrics)
+            self.assertIn("memory", collected_metrics[name])
+            self.assertIn("cur_mem", collected_metrics[name]["memory"])
+            self.assertIn("max_mem", collected_metrics[name]["memory"])
+            self.assertEqual(num_summarization_values, len(collected_metrics[name]["memory"]["cur_mem"]))
+            self.assertEqual(num_summarization_values, len(collected_metrics[name]["memory"]["max_mem"]))
+
+            self.assertListEqual(generate_metric_list(memory_usage_values),
```

Comment:
Can you add another test where you actually assert concrete values that got computed? This way we can check the validity of the computations as well. I'm mostly concerned about division.

Reply:
Yup. I am adding that in the test_extension test, which will do the entire extension_handler.run() and then do a poll_tracked_list. That unittest will go in another PR.